src/exchanges/coinex-client.js

```diff
@@ -0,0 +1,208 @@
+const moment = require("moment");
+const BasicClient = require("../basic-client");
+const BasicMultiClient = require("../basic-multiclient");
+const Ticker = require("../ticker");
+const Trade = require("../trade");
+const Level2Point = require("../level2-point");
+const Level2Snapshot = require("../level2-snapshot");
+const Level2Update = require("../level2-update");
+
+class CoinexClient extends BasicMultiClient {
+  constructor() {
+    super(CoinexSingleClient);
+
+    // Do we need these properties here???
+    this.hasTickers = true;
+    this.hasTrades = true;
+    this.hasLevel2Snapshots = false;
+    this.hasLevel2Updates = true;
+    this.hasLevel3Updates = false;
+  }
+}
+
+class CoinexSingleClient extends BasicClient {
+  constructor() {
+    super("wss://socket.coinex.com/", "Coinex");
+    this.hasTickers = true;
+    this.hasTrades = true;
+    this.hasLevel2Snapshots = false;
+    this.hasLevel2Updates = true;
+    this.hasLevel3Updates = false;
+  }
+
+  _sendSubTicker(remote_id) {
+    this._wss.send(
+      JSON.stringify({
+        method: "state.subscribe",
+        params: [remote_id],
+        id: 1,
+      })
+    );
+  }
+
+  _sendUnsubTicker() {
+    this._wss.send(
+      JSON.stringify({
+        method: "state.unsubscribe",
+      })
+    );
+  }
+
+  _sendSubTrades(remote_id) {
+    this._wss.send(
+      JSON.stringify({
+        method: "deals.subscribe",
+        params: [remote_id],
+        id: 1,
+      })
+    );
+  }
+
+  _sendUnsubTrades() {
+    this._wss.send(
+      JSON.stringify({
+        method: "deals.unsubscribe",
+      })
+    );
+  }
+
+  _sendSubLevel2Updates(remote_id) {
+    this._wss.send(
+      JSON.stringify({
+        method: "depth.subscribe",
+        params: [remote_id, 100, "0"], // 100 is the maximum number of items Coinex will let you request
+        id: 1,
+      })
+    );
+  }
+
+  _sendUnsubLevel2Updates() {
+    this._wss.send(
+      JSON.stringify({
+        method: "depth.unsubscribe",
+      })
+    );
+  }
+
+  _onMessage(raw) {
+    let msg = JSON.parse(raw);
+
+    let { method, params } = msg;
+
+    // if params is defined, then this is a response to an event that was subscribed to, otherwise this is the initial connection response
+    if (params) {
```

Comment:
minor code structure thing, but may be cleaner to remove indent for the whole block by just doing 
```
if(!params) return;
```